src/Microsoft.Health.Dicom.Core/Features/Update/UpdateInstanceService.cs

```diff
@@ -0,0 +1,79 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Extensions;
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Core.Features.Context;
+using Microsoft.Health.Dicom.Core.Features.Operations;
+using Microsoft.Health.Dicom.Core.Features.Routing;
+using Microsoft.Health.Dicom.Core.Models.Operations;
+using Microsoft.Health.Dicom.Core.Models.Update;
+using Microsoft.Health.Operations;
+
+namespace Microsoft.Health.Dicom.Core.Features.Update;
+
+public class UpdateInstanceService : IUpdateInstanceService
+{
+    private readonly IGuidFactory _guidFactory;
+    private readonly IDicomOperationsClient _client;
+    private readonly IUrlResolver _urlResolver;
+    private readonly IDicomRequestContextAccessor _contextAccessor;
+
+    private static readonly OperationQueryCondition<DicomOperation> Query = new OperationQueryCondition<DicomOperation>
+    {
+        Operations = new DicomOperation[] { DicomOperation.Update },
+        Statuses = new OperationStatus[]
+        {
+            OperationStatus.NotStarted,
+            OperationStatus.Running,
+        }
+    };
+
+    public UpdateInstanceService(
+        IGuidFactory guidFactory,
+        IDicomOperationsClient client,
+        IUrlResolver iUrlResolver,
+        IDicomRequestContextAccessor contextAccessor)
+    {
+        EnsureArg.IsNotNull(guidFactory, nameof(guidFactory));
+        EnsureArg.IsNotNull(client, nameof(client));
+        EnsureArg.IsNotNull(iUrlResolver, nameof(iUrlResolver));
+        EnsureArg.IsNotNull(contextAccessor, nameof(contextAccessor));
+
+        _guidFactory = guidFactory;
+        _client = client;
+        _urlResolver = iUrlResolver;
+        _contextAccessor = contextAccessor;
+    }
+
+    public async Task<OperationReference> QueueUpdateOperationAsync(
+        UpdateSpecification spec,
+        CancellationToken cancellationToken = default)
+    {
+        EnsureArg.IsNotNull(spec, nameof(spec));
+        EnsureArg.IsNotNull(spec.ChangeDataset, nameof(spec.ChangeDataset));
+
+        UpdateRequestValidator.ValidateDicomDataset(spec.ChangeDataset);
```

Comment:
Are we missing to validate if the items are in the known tag bucket?

Reply:
Added the known tag bucket in `UpdateTags.cs`